ur_bringup/launch/ur_control.launch.py

```diff
@@ -125,6 +133,30 @@ def launch_setup(context, *args, **kwargs):
             "use_tool_communication:=",
             use_tool_communication,
             " ",
+            "tool_parity:=",
+            tool_parity,
+            " ",
+            "tool_baud_rate:=",
+            tool_baud_rate,
+            " ",
+            "tool_stop_bits:=",
+            tool_stop_bits,
+            " ",
+            "tool_rx_idle_chars:=",
+            tool_rx_idle_chars,
+            " ",
+            "tool_tx_idle_chars:=",
+            tool_tx_idle_chars,
+            " ",
+            "tool_device_name:=",
+            tool_device_name,
+            " ",
+            "tool_tcp_port:=",
+            tool_tcp_port,
+            " ",
+            "tool_voltage:=",
+            tool_voltage,
+            " ",
```

Comment:
The parameters here are leading to "nirvana" (read nowhere) :smiley: 

You would probably like to add them into `ur.urdf.xacro`, `ur_macro.xacro` and `ur.ros2_control.xacro`.

Or I am missing something here?

I think that your example works because you are using TCP tool, but if you start to use RS485 tool, this would not be configured.

Reply:
Aren't they read [here](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/a0cfee50ae7b9318ad705d9634716a76cdb131a7/ur_description/urdf/ur.ros2_control.xacro?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L35-L42)?

Reply:
Yes, sorry. All good. I was looking to the wrong file.